src/server/admin/server/api_server.go

```diff
@@ -257,6 +263,88 @@ func (a *apiServer) Extract(request *admin.ExtractRequest, extractServer admin.A
 			}
 		}
 	}
+	if !request.NoEnterprise {
+		state, err := pachClient.Enterprise.GetState(pachClient.Ctx(), &enterprise.GetStateRequest{})
+		if err != nil {
+			return err
+		}
+		// Don't write out expired or missing enterprise keys
+		if state.State == enterprise.State_ACTIVE {
+			if err := writeOp(&admin.Op{Op1_12: &admin.Op1_12{ActivateEnterprise: &enterprise.ActivateRequest{ActivationCode: state.ActivationCode}}}); err != nil {
+				return err
+			}
+		} else {
+			logrus.Warnf("Enterprise license state: %v, not extracting", state.State)
+		}
+	}
+
+	// Once all the repos and pipelines have been restored, then try to enable auth. Our auth checks block creating a pipeline if the output repo already exists,
```

Comment:
I remember Yusuf talking about this with build pipelines, and while I don't think there's nothing unique to auth that requires  this check (so we could get rid of it without breaking anything new, I think), it actually strikes me as a bug that we don't check this in all cases.

Creating a pipeline when an existing output repo exists seems broken even without auth. What if you create two pipelines with the same name and they're writing on top of each other? This is maybe a JD question.

Reply:
> What if you create two pipelines with the same name and they're writing on top of each other?

Things like the PipelineInfo collection seem to be keyed on the pipeline name, so I don't believe it's possible to have two pipelines with the same name.

Reply:
That makes sense. FWIW this check is in `authorizePipelineOpInTransaction`. All we have to do is disable it is change the check for `Create` so that it require either a) the output repo doesn't exist or b) the user has WRITER access to it.